core/modules/image/image.module

```diff
@@ -1349,3 +1359,186 @@ function image_filter_keyword($value, $current_pixels, $new_pixels) {
   }
   return (int) $value;
 }
+
+/**
+ * Preprocess function for theme_image_style().
+ *
+ * If theme_image_style() is called without setting explicit height/width for
+ * an SVG image, attempt to set those dimensions. This helps in particular when
+ * uploading an image for the first time in a content form.
+ */
+function image_preprocess_image_style(&$variables) {
+  $uri = $variables['path'];
+  // Only try to get file information if $uri is not empty.
+  if (!empty($uri) && image_is_svg($uri)) {
+    if (empty($variables['width']) && empty($variables['height'])) {
+      if ($dimensions = image_get_svg_dimensions($uri)) {
+        $variables['width'] = $dimensions['width'];
+        $variables['height'] = $dimensions['height'];
+      }
+      else {
+        $variables['width'] = IMAGE_SVG_DEFAULT_WIDTH;
+        $variables['height'] = IMAGE_SVG_DEFAULT_HEIGHT;
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_file_presave().
+ *
+ * Set the dimensions in the "metadata" property used by the File Entity module.
+ */
+function image_file_presave($file) {
+  if (image_is_svg($file->uri) && isset($file->metadata)) {
+    $svg_dimensions = image_get_svg_dimensions($file->uri);
+    if ($svg_dimensions) {
+      $file->metadata['width'] = $svg_dimensions['width'];
+      $file->metadata['height'] = $svg_dimensions['height'];
+    }
+  }
+}
+
+
+/**
+ * Implements hook_field_attach_presave().
```

Comment:
This feels intensive - like it was a contrib module that couldn't change core in a better way. Isn't there a way to set the width and height in the image or file element earlier? Instead of having to find image fields this way.

Reply:
In progress, but removed that `hook_field_attach_presave()` implementation, now that we get image dimensions from `image_get_info()`